pydhsfw/tcpip.py

```diff
@@ -247,7 +247,7 @@ def _disconnect(self):
                 self._stream_reader.socket = None
                 self._stream_writer.socket = None
                 if sock:
-                    sock.shutdown(socket.SHUT_RDWR)
+                    #sock.shutdown(socket.SHUT_RDWR)
                     sock.close()
```

Comment:
https://stackoverflow.com/questions/409783/socket-shutdown-vs-socket-close

There are scenarios where `close()` will not work due to reference counting. In this scenario just a `close()` appears to be working, but other cases may not work. Generally we really do want to shutdown the socket nicely so the other end has an opportunity to gracefully disconnect and be available to us sooner for reconnecting. Our local socket reference counting will eventually sort itself out through garbage collection. Better to test the socket status prior to calling `shutdown()` or wrap it in a try/except explicitly catching `OSError: [Errno 57] Socket is not connected`.

Reply:
Not sure how to test socket status so I added a try/except/finally block.